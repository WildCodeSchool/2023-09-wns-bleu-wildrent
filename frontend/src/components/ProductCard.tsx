import { ProductCardProps } from '../types/index';
import Link from 'next/link';
import React from 'react';

function ProductCard({ productRef: { name, image, priceHT }, link }: ProductCardProps) {
  return (
    <Link href={link}>
      <div className="card card-compact w-96 bg-base-100 shadow-xl">
        <figure>
          <img src={image} alt={name} />
        </figure>
        <div className="card-body">
          <h2 className="card-title text-center">{name}</h2>
          <div className="card-price text-right">{priceHT} â‚¬ / jour</div>
        </div>
      </div>
    </Link>
  );
}

export default ProductCard;
