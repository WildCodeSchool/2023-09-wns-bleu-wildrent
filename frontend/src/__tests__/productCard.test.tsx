import ProductCard from '@/components/ProductCard';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

describe('ProductCard Component', () => {
  it('renders correctly', () => {
    const productRef = { id: 1, name: 'Guirlande', image: 'https://img.com/i.png', priceHT: 1 };
    const view = render(<ProductCard productRef={productRef} link="/products/1"></ProductCard>);
    const link = screen.getByRole('link');
    const heading = screen.getByRole('heading', { level: 2 });
    const image = screen.getByRole('img');

    expect(link).toHaveAttribute('href', '/products/1');
    expect(heading).toBeInTheDocument();
    expect(screen.getByText('Guirlande')).toBeInTheDocument();
    expect(image).toHaveAttribute('src', productRef.image);

    expect(view.baseElement).toMatchSnapshot();
  });
});
