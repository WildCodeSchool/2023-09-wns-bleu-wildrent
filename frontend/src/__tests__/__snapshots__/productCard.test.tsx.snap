// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Component renders correctly 1`] = `
<body>
  <div>
    <a
      href="/products/1"
    >
      <div
        class="card card-compact w-96 bg-base-100 shadow-xl"
      >
        <figure>
          <img
            alt="Guirlande"
            src="https://img.com/i.png"
          />
        </figure>
        <div
          class="card-body"
        >
          <h2
            class="card-title text-center"
          >
            Guirlande
          </h2>
          <div
            class="card-price text-right"
          >
            1
             â‚¬ / jour
          </div>
        </div>
      </div>
    </a>
  </div>
</body>
`;
